tests/testthat/test-universal_useful.R

```diff
@@ -81,3 +81,61 @@ test_that("find_by_class works", {
     find_by_class("numeric", UU_testing_env),
     "Could not find object with class numeric. Has it been instantiated?")
 })
+
+
+test_that("match_letters function matches letters correctly", {
+  # Test case 1: Single letter match
+  result_1 <- match_letters("apple", "a", n = 1)
+  expect_equal(result_1, "a")
+
+  # Test case 2: Case insensitive match
+  result_2 <- match_letters(c("apple", "banana"), "A", ignore.case = TRUE)
+  expect_equal(result_2, "A")
+
+  # Test case 3: Capitalize matched letters
+  result_3 <- match_letters(c("apple", "banana"), "a", capitalize = TRUE)
+  expect_equal(result_3, "A")
+})
+
+
+test_that("missing_args function identifies missing arguments correctly", {
+  # Test case 1: No missing arguments when all arguments are passed
+  fn1 <- function(a, b) missing_args()
+  result_1 <- fn1(a = 1, b = 2)
+  print(result_1)
+  expect_length(result_1, 0)
+  expect_equal(result_1, character(0))
+
+  # Test case 2: Identify missing arguments when some arguments are not passed
+  fn2 <- function(a, b) missing_args()
+  result_2 <- fn2(a = 1)
+  expect_length(result_2, 1)
+  expect_equal(result_2, "b")
+
+  # Test case 3: Include NULL arguments in missing arguments
+  fn3 <- function(a = NULL, b) missing_args()
+  result_3 <- fn3()
+  expect_length(result_3, 2)
+  expect_equal(result_3, c("a", "b"))
+
+  # Test case 4: Exclude default arguments from missing arguments
+  fn4 <- function(a = 1, b) missing_args()
+  result_4 <- fn4(b = 2)
+  expect_length(result_1, 0)
+  expect_equal(result_1, character(0))
+})
```

Comment:
Can you check for this set of tests? I'm having some weird problems:

- When I run them individually (i.e., line by line) they all pass.
- When I execute this chunk of code, some of them pass.
- When I run the whole unit test, none of this tests pass.

I'm wondering if it is some issue with the function regarding the environment used by testthat package.

Reply:
@teotenn There was indeed an issue with the `corresponding_call` default value, I think I've fixed it now as the test are running correctly in all three cases now.